tutorials/sns-instances-notification-system/index.mdx

```diff
@@ -0,0 +1,301 @@
+    project_id = scaleway_account_project.sns_tutorial.id
+    name = "suscriber-server"
+    type  = "PLAY2-PICO"
+    image = "debian_bookworm"
+    # private_network {
+    #pn_id = scaleway_vpc_private_network.sns_tutorial_pn.id
+    #}
+    ip_id = scaleway_instance_ip.subscriber_public_ip.id
+    security_group_id= scaleway_instance_security_group.sns_www.id
+
+    # User data to run the server at start-up
+    user_data = {
+      cloud-init = <<-EOF
+      #cloud-config
+      runcmd:
+        - |
+          apt-get update && apt-get install -y docker.io
+          systemctl start docker
+          systemctl enable docker
+          docker pull rg.fr-par.scw.cloud/sns-x-instance-tutorial/subscriber-server:1.0
+          docker run -d --restart=always \
+            --name subscriber-server \
+            -p 8081:8081 \
+            rg.fr-par.scw.cloud/sns-x-instance-tutorial/subscriber-server:1.0
+    EOF
+    }
+  }
+  ```
+
+3. Create a `terraform_data` resource to wait for the port 8081 to be opened on the subscriber Instance before subscribing to the topic we previously created:
+  ```
+  resource "terraform_data" "bootstrap" {
+    triggers_replace = [
+      scaleway_instance_server.subscriber_sns_tuto_instance.id,
+    ]
+
+    # Wait for 8081 port to be opened with a 3 minutes timeout
+    provisioner "local-exec" {
+      command = <<EOF
+      #!/bin/bash
+      TIMEOUT=180
+      START_TIME=$(date +%s)
+      while [ $(($(date +%s) - $START_TIME)) -lt $TIMEOUT ]; do
+        nc -z ${scaleway_instance_server.subscriber_sns_tuto_instance.public_ip} 8081 && exit 0
+        sleep 1
+      done
+      echo "Timeout reached"
+      exit 1
+      EOF
+    }
+  }
+  ```
+This will allow the subscriber Instance to immediately receive the [confirmation link](/serverless/messaging/how-to/create-manage-subscriptions/#how-to-confirm-an-http-or-https-subscription) to subscribe to the topic, and not wait for the retry policy to confirm the subscription.
+The confirmation step is done manually for the purposes of this tutorial, but it could also be performed with an HTTP request.
+
+4. Create the SNS topic subscription with the subscriber Instance's endpoint:
+  ```
+  resource scaleway_mnq_sns_topic_subscription main {
+    project_id = scaleway_account_project.sns_tutorial.id
+    access_key = scaleway_mnq_sns_credentials.main.access_key
+    secret_key = scaleway_mnq_sns_credentials.main.secret_key
+    topic_id = scaleway_mnq_sns_topic.topic.id
+    protocol = "http"
+    endpoint = "http://${scaleway_instance_ip.subscriber_public_ip.address}:8081/notifications"
+  }
+  ```
+
+5. Create the publisher Instance's IP address:
+  ```
+  resource "scaleway_instance_ip" "publisher_public_ip" {
+    project_id = scaleway_account_project.sns_tutorial.id
+    zone = "fr-par-1"
+  }
+  ```
+
+6. Create the publisher Instance itself, with all the environment variables it will need to publish the notifications to the topic:
```

Comment:
```suggestion
7. Create the publisher Instance itself, with all the environment variables it will need to publish the notifications to the topic:
```